system-tests-new/fortran-prebuild/test_fortran_prebuild.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import zlib
+from pathlib import Path
+from unittest import mock
+
+from fab.util import file_walk
+
+from fab.build_config import BuildConfig
+from fab.steps.analyse import Analyse
+from fab.steps.compile_fortran import CompileFortran
+from fab.steps.grab import GrabFolder, GrabPreBuild
+from fab.steps.link_exe import LinkExe
+from fab.steps.preprocess import fortran_preprocessor
+from fab.steps.walk_source import FindSourceFiles
+
+
+class TestFortranPrebuild(object):
+
+    def build_config(self, fab_workspace, prebuild_folder=None):
+        logging.getLogger('fab').setLevel(logging.WARNING)
+
+        build_config = BuildConfig(
+            project_label='test_prebuild',
+            fab_workspace=fab_workspace,
+            steps=[
+                GrabFolder(Path(__file__).parent / 'project-source', dst='src'),
+                *([GrabPreBuild(prebuild_folder)] if prebuild_folder else []),
+                FindSourceFiles(),
+                fortran_preprocessor(preprocessor='cpp -traditional-cpp -P'),
+                Analyse(root_symbol='my_prog'),
+                CompileFortran(compiler='gfortran -c', common_flags=['-J', '$output']),
```

Comment:
Good question. @MatthewHambley has also mentioned this. I think we're going to have a chat about this sometime soon. Would be good to have you there.

I think Fab might become compiler-aware soon, so it would know how to use gfortran to compile a file. If that happens, all strings like this will probably be replaced.

FWIW, the original thinking was Q: _"what's the command to just compile a file"_ A: _"gfortran -c"_.